### PR TITLE
feat : Improve workspace archiving experience : ask a confirmation + fix refresh when creating /archiving a workspace

### PR DESCRIPTION
**Pull request by YolanFery**
Creating a workspace was not refreshing the cache, added that

Archiving a workspace is an major action -> provide a  name validation before confirming archiving 

## Changes

- Translations
- Workspace name validation in the dialog for archiving
- Clear `workspaces` cache on create

## How/what to test

- Adding and removing workspaces is working seamlessly 

## Screenshots / screencast

Screenshot 2024-12-06 at 16 42 27

### CODE REVIEW

File: src/core/components/Dialog.tsx [hunk 1]
```diff
@@ -13,6 +14,7 @@ import {
 type DialogProps = {
   open: boolean;
   onClose: (value: any) => void;
+  onClick?: MouseEventHandler<HTMLDivElement>;
```
Comment (YolanFery) [inline src/core/components/Dialog.tsx:17]: Alternatively we might want to expose/extend the props of the underlying HTML element

File: src/workspaces/features/SidebarMenu/SidebarMenu.tsx [hunk 2]
```diff
@@ -224,6 +234,28 @@ const SidebarMenu = (props: SidebarMenuProps) => {
                   <span className="text-sm leading-tight tracking-tight">
                     {ws.name}
                   </span>
+                  {ws.permissions.delete && (
```
Comment (YolanFery) [inline src/workspaces/features/SidebarMenu/SidebarMenu.tsx:237]: Is this permission enough ? This is one currently used in the settings panel but it seems weird that we only check for the workspace permissions and not the user permission. I lack some context here

File: src/workspaces/features/ArchiveWorkspaceDialog/ArchiveWorkspaceDialog.tsx [hunk 3]
```diff
@@ -54,20 +55,30 @@ const ArchiveWorkspaceDialog = (props: ArchiveWorkspaceDialogProps) => {
   };
 
   return (
-    <Dialog open={open} onClose={onClose}>
+    <Dialog open={open} onClose={onClose} onClick={(e) => e.stopPropagation()}>
```
Comment (qgerome) [inline src/workspaces/features/ArchiveWorkspaceDialog/ArchiveWorkspaceDialog.tsx:58]: What's the purpose of the onClick ?

File: src/workspaces/features/ArchiveWorkspaceDialog/ArchiveWorkspaceDialog.tsx [hunk 4]
```diff
@@ -54,20 +55,30 @@ const ArchiveWorkspaceDialog = (props: ArchiveWorkspaceDialogProps) => {
...
       <Dialog.Title>
         {t("Archive {{name}}", { name: workspace.name })}
       </Dialog.Title>
       <Dialog.Content className="space-y-4">
         <p>
           {t("You're about to archive this workspace and all its content.")}
         </p>
+        <p>{t("Please enter the workspace name to confirm:")}</p>
```
Comment (qgerome) [inline src/workspaces/features/ArchiveWorkspaceDialog/ArchiveWorkspaceDialog.tsx:66]: I would at least put the name of the workspace in the sentence. Bonus: having it as a placeholder of the input below

Comment (YolanFery) [reply src/workspaces/features/ArchiveWorkspaceDialog/ArchiveWorkspaceDialog.tsx:69]: Good idea, thanks, change dit

File: src/workspaces/features/ArchiveWorkspaceDialog/ArchiveWorkspaceDialog.tsx (hunk 3 above)
Comment (YolanFery) [reply src/workspaces/features/ArchiveWorkspaceDialog/ArchiveWorkspaceDialog.tsx]: It was necessary on previous revisions to keep the dialog open but now it seems its not needed anymore so I removed, thanks

File: src/workspaces/features/ArchiveWorkspaceDialog/ArchiveWorkspaceDialog.tsx (hunk 4 above)
Comment (qgerome) [reply src/workspaces/features/ArchiveWorkspaceDialog/ArchiveWorkspaceDialog.tsx:69]: don't know If you are aware of it but we have a nice `<Trans/>` component that allow you to translate jsx/html (and thus, keep styles to put some items in emphase)

**Review by qgerome — APPROVED** [review]

File: src/workspaces/features/ArchiveWorkspaceDialog/ArchiveWorkspaceDialog.tsx (hunk 4 above)
Comment (qgerome) [reply src/workspaces/features/ArchiveWorkspaceDialog/ArchiveWorkspaceDialog.tsx:69]: The logic is that it takes the raw string and replace all tags by numeroted placeholders <0></0> and do the interpolation when it displays the translation. Pretty neat.

**Pull request merged**